spec/aleph/newman/Aleph.postman_collection.json

```
@@ -0,0 +1,373 @@
+			"event": [
+				{
+					"listen": "test",
+					"script": {
+						"id": "e77b37a0-6d95-4688-8aff-d1e7ffc12893",
+						"exec": [
+							"pm.test(\"Response time is less than 1000ms\", function () {",
+							"    pm.expect(pm.response.responseTime).to.be.below(1000);",
+							"});",
+							"pm.test(\"Response time is less than 200ms\", function () {",
+							"    pm.expect(pm.response.responseTime).to.be.below(200);",
+							"});",
+							"pm.test(\"Body matches string\", function () {",
+							"    pm.expect(pm.response.text()).to.include(\"ND Full Catalog\");",
+							"});"
+						],
+						"type": "text/javascript"
+					}
+				}
+			],
+			"request": {
+				"method": "GET",
+				"header": [],
+				"url": {
+					"raw": "https://{{AlephURL}}/F/?func=find-b-0&local_base=ndu01pub",
+					"protocol": "https",
+					"host": [
+						"{{AlephURL}}"
+					],
+					"path": [
+						"F",
+						""
+					],
+					"query": [
+						{
+							"key": "func",
+							"value": "find-b-0"
+						},
+						{
+							"key": "local_base",
+							"value": "ndu01pub"
+						}
+					]
+				}
+			},
+			"response": []
+		},
+		{
+			"name": "Check for OPAC Record",
+			"event": [
+				{
+					"listen": "test",
+					"script": {
+						"id": "e5ebb3e5-f206-49be-aa77-a246f4155c48",
+						"exec": [
+							"pm.test(\"Status code is 200\", function () {",
+							"    pm.response.to.have.status(200);",
+							"});",
+							"pm.test(\"Response time is less than 1000ms\", function () {",
+							"    pm.expect(pm.response.responseTime).to.be.below(1000);",
+							"});",
+							"pm.test(\"Body matches string\", function () {",
+							"    pm.expect(pm.response.text()).to.include(\"LF 509\");",
+							"});"
+						],
+						"type": "text/javascript"
+					}
+				}
+			],
+			"request": {
+				"method": "GET",
+				"header": [],
+				"url": {
+					"raw": "https://{{AlephURL}}/F/1Q3EMSL9NCKB6P19QU5QRKMCY4QNLBDCUTADR3LJS7YL33JEK5-08833?func=direct&doc_number=000123456",
```

Looks like the URL has a session ID in it (the string following F/). You might need to eliminate this, or if a session ID is needed to connect with an earlier call, should it use a variable?

---

We can make those changes if needed - is there something in the session ID that would cause the tests to fail or succeed unexpectedly? Such as moving between systems (such as from `alephprod` to `alephpprd`?

---

The session ID is randomly generated when you start a web interaction, and then they expire. I don't remember how long they last -- perhaps 24 or 48 hrs, or something. I'm not sure what will happen if you pass a session ID that no longer exists with an API call.

---

@abbales Do you think an adequate substitution here would be to make a request with a bad session ID and test against the expected results of that?

We could probably also specifically request a Session ID and use it later, but that would be more involved. Thoughts?

---

Re-wrote the spec and requested a new review from @abbales